rfd/0136-modern-signature-algorithms.md

```diff
@@ -0,0 +1,881 @@
+    ECDSA support is widespread enough I don't think we should support bigger
+    RSA keys, we should guide people toward ECDSA instead.
+
+#### Host CA
+
+keys: ssh, tls
+
+uses: host ssh cert signing, host tls cert signing, ssh clients trust this CA
+
+* current SSH algo: `RSA2048_PKCS1_SHA512`
+* proposed supported SSH algos:
+  * `Ed25519`
+  * `ECDSA_P256_SHA256`
+  * `RSA2048_PKCS1_SHA512`
+  * `RSA3072_PKCS1_SHA512`
+  * `RSA4096_PKCS1_SHA512`
+* proposed supported SSH `allowed_subject_algorithms`:
+  * `Ed25519`
+  * `ECDSA_P256_SHA256`
+  * `RSA2048_PKCS1_SHA512`
+  * `RSA3072_PKCS1_SHA512`
+  * `RSA4096_PKCS1_SHA512`
+* reasoning:
+  * `Ed25519` is the current best-in-class for SSH
+  * `ECDSA_P256_SHA256` has Go BoringCrypto support
+  * some environments still require RSA
+
+* current TLS algo: `RSA2048_PKCS1_SHA256`
+* proposed supported TLS algos:
+  * `ECDSA_P256_SHA256`
+  * `Ed25519`
+  * `RSA2048_PKCS1_SHA256`
+* proposed supported TLS `allowed_subject_algorithms`:
+  * `ECDSA_P256_SHA256`
+  * `Ed25519`
+  * `RSA2048_PKCS1_SHA256`
+* reasoning:
+  * `EDCSA_P256_SHA256` has BoringCrypto support.
+  * `Ed25519` support is becoming more common and some people prefer it
+  * `RSA2048_PKCS1_SHA256` will be supported for backward compatibility
+
+#### Database CA
+
+keys: tls
+
+uses:
+
+* signs (often) long-lived db cert used to authenticate db to database service
+* signs short-lived proxy cert used to authenticate proxy to database service
+* signed snowflake JWTs
+
+* current TLS algo: `RSA2048_PKCS1_SHA256`
+* proposed supported TLS algos:
+  * `ECDSA_P256_SHA256`
+  * `Ed25519`
+  * `RSA2048_PKCS1_SHA256`
+  * `RSA3072_PKCS1_SHA256`
+  * `RSA4096_PKCS1_SHA256`
+* proposed supported TLS `allowed_subject_algorithms`:
+  * `ECDSA_P256_SHA256`
+  * `Ed25519`
+  * `RSA2048_PKCS1_SHA256`
+  * `RSA3072_PKCS1_SHA256`
+  * `RSA4096_PKCS1_SHA256`
+* reasoning:
+  * some database protocols still require RSA, reduce friction by keeping it as
+    the default
+  * eventually we should default to `RSA3072_PKCS1_SHA256` for long-lived db
+    certs that require RSA, but use an algorithm that is cheaper to generate
+    keys for the proxy certs that authenticate to the db service, since these
+    are only used internally we should go for Ed25519
+
+#### OpenSSH Host CA
+
+keys: ssh
+
+uses: signs user certs to authenticate to registered OpenSSH nodes, registered
+OpenSSH nodes trust this CA.
+
+* current SSH algo: `RSA2048_PKCS1_SHA512`
+* proposed supported SSH algos:
+  * `Ed25519`
+  * `ECDSA_P256_SHA256`
+  * `RSA2048_PKCS1_SHA512`
+  * `RSA3072_PKCS1_SHA512`
+  * `RSA4096_PKCS1_SHA512`
+* proposed supported SSH `allowed_subject_algorithms`:
+  * `Ed25519`
+  * `ECDSA_P256_SHA256`
+  * `RSA2048_PKCS1_SHA512`
+* reasoning:
+  * `Ed25519` is the current best-in-class for SSH
+  * `ECDSA_P256_SHA256` has Go BoringCrypto support
+  * some environments still require RSA
+
+#### JWT CA
```

Comment:
The JWT, OIDC and SAML keys kind of terrify me a bit, but I guess that changing the algorithm and rekeying them if one encounters problems with some third party app/service is relatively low-effort.

@mdwn do you know if ECDSA support is widespread enough in the SAML ecosystem that we can default to it?